src/mono/CMakeLists.txt

```diff
@@ -358,6 +358,11 @@ if(NOT "${MSVC_C_ARCHITECTURE_ID}" STREQUAL "")
   set(CMAKE_SYSTEM_PROCESSOR "${MSVC_C_ARCHITECTURE_ID}")
 endif()
 
+if(HOST_BROWSER OR HOST_WASI)
+  # CMAKE_SYSTEM_PROCESSOR is set to x86 by emscripten
```

Comment:
This is a workaround for an emscripten problem, so it might not be needed with wasi.